zone/guild_mgr.cpp

```diff
@@ -1454,3 +1475,338 @@ bool GuildBankManager::AllowedToWithdraw(uint32 GuildID, uint16 Area, uint16 Slo
+	ServerOP_GuildMessage_Struct* s_out = (ServerOP_GuildMessage_Struct*)s_outapp->pBuffer;
+
+    s_out->guild_id = guild_id;
+	strn0cpy(s_out->player_name, player_name.c_str(), sizeof(s_out->player_name));
+
+    worldserver.SendPacket(s_outapp);
+	safe_delete(s_outapp);
+}
+
+void ZoneGuildManager::SendToWorldMemberAdd(uint32 guild_id, uint32 char_id, uint32 level, uint32 _class, uint32 rank, uint32 zone_id, std::string player_name)
+{
+	auto s_outapp = new ServerPacket(ServerOP_GuildMemberAdd, sizeof(ServerOP_GuildMessage_Struct));
+	auto s_out = (ServerOP_GuildMessage_Struct*)s_outapp->pBuffer;
+
+    s_out->guild_id       = guild_id;
+	s_out->player_level   = level;
+	s_out->player_rank    = rank;
+	s_out->player_zone_id = zone_id;
+	s_out->player_class   = _class;
+	strn0cpy(s_out->player_name, player_name.c_str(), sizeof(s_out->player_name));
+
+    worldserver.SendPacket(s_outapp);
+	safe_delete(s_outapp);
+}
+
+void ZoneGuildManager::SendToWorldSendGuildList()
+{
+	auto s_outapp = new ServerPacket(ServerOP_GuildSendGuildList, sizeof(ServerOP_GuildMessage_Struct));
+	auto s_out = (ServerOP_GuildMessage_Struct*)s_outapp->pBuffer;
+
+	worldserver.SendPacket(s_outapp);
+	safe_delete(s_outapp);
+}
+
+bool ZoneGuildManager::RemoveMember(uint32 guild_id, uint32 char_id, std::string player_name)
+{
+	GuildMembersRepository::DeleteOne(*m_db, char_id);
+	SendToWorldMemberRemove(guild_id, player_name);
+	return true;
+}
+
+void ZoneGuildManager::MemberAdd(uint32 guild_id, uint32 char_id, uint32 level, uint32 _class, uint32 rank, uint32 zone_id, std::string player_name)
+{
+	GuildMembersRepository::GuildMembers m;
+	m.alt            = 0;
+	m.banker         = rank == GUILD_LEADER ? 1 : 0;
+	m.last_tribute   = 0;
+    m.total_tribute  = 0;
+    m.tribute_enable = 0;
+    m.rank_          = rank;
+    m.char_id        = char_id;
+    m.guild_id       = guild_id;
+    m.public_note.clear();
+
+	GuildMembersRepository::ReplaceOne(*m_db, m);
+	UpdateDbMemberOnline(char_id, true);
+	SendToWorldMemberAdd(guild_id, char_id, level, _class, rank, zone_id, player_name);
+}
+
+bool ZoneGuildManager::IsActionABankAction(GuildAction action)
+{
+	return action == GUILD_ACTION_BANK_DEPOSIT_ITEMS ||
+		   action == GUILD_ACTION_BANK_PROMOTE_ITEMS ||
+		   action == GUILD_ACTION_BANK_VIEW_ITEMS ||
+		   action == GUILD_ACTION_BANK_WITHDRAW_ITEMS;
+}
+
+void ZoneGuildManager::SendToWorldMemberRankUpdate(uint32 guild_id, uint32 rank, uint32 banker, uint32 alt, bool no_update, const char *player_name)
+{
+	auto outapp = new ServerPacket(ServerOP_GuildRankUpdate, sizeof(ServerGuildRankUpdate_Struct));
+	auto sr     = (ServerGuildRankUpdate_Struct *) outapp->pBuffer;
+
+	sr->guild_id  = guild_id;
+	sr->rank      = rank;
+	sr->banker    = banker;
+	sr->alt       = alt;
+	sr->no_update = no_update;
+	strn0cpy(sr->member_name, player_name, sizeof(sr->member_name));
+
+	worldserver.SendPacket(outapp);
+	safe_delete(outapp)
```

Comment:
Missing `;`.

Reply:
I removed it as it is within the macro and CLion highlights it to be removed.

Reply:
I have put them all back.  Anything else, please let me know.